### PR TITLE
PROGRAMMES-6287 related topics and programmes

### PR DESCRIPTION
**Pull request by benolot**
Ticket: https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6287

Tested it on: http://www.bbc.co.uk/programmes/b071vl2l

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by votemike** [comment]
Could you test this on small screen sizes. It doesn't look quite right.
screen shot 2018-04-19 at 16 13 03

File: src/ExternalApi/Ada/Service/AdaClassService.php [hunk 1]
```diff
@@ -128,6 +129,26 @@ private function buildRequestUrl(
...
     }
 
+    public function findSuggestedByProgrammeItem(Programme $programme)
+    {
+        $relatedByTag = $this->requestRelatedProgrammeItems((string)$programme->getPid(), 'tag', null, 2, 'random', null, 1);
+        $relatedByBrand = $this->requestRelatedProgrammeItems((string)$programme->getPid(), null, $programme->getTleo()->getPid(), 2, 'random', null, 1);
+        $relatedByCategory = $this->requestRelatedProgrammeItems((string)$programme->getPid(), null, null, 2, 'random', null, 5);
+
+        //@TODO remove
+        $unwrapped = \GuzzleHttp\Promise\unwrap(['a' => $relatedByTag, 'b' => $relatedByBrand, 'c' => $relatedByCategory]);
```
[the remark this answers is not in the document]
Comment (benolot) [reply src/ExternalApi/Ada/Service/AdaClassService.php]: That was a commented out section for debugging, I've now removed it in an incoming Commit

**Comment by BjornTwachtmann** [comment]
Okay. A few comments on the structure of this. Sorry. I usually go through this kind of stuff in person rather than writing a load of guff on PRs so that we can both understand it, but...

Just FYI I really don't understand ADA that well, so if any of this is rubbish due to my ignorance of what it's doing, do just call me on it. 

* I don't like that the ProgrammeItems from the database are queried separately in the controller. If they're required I think the ADA service should query them. This is what the recommendations service does, for example, see https://github.com/bbc/programmes-frontend/blob/master/src/ExternalApi/RecEng/Service/RecEngService.php#L108 . It can be done in the callback when the promise is resolved.
* There seems to be some gymnastics going on in RelatedProgrammesPresenter to relate an AdaProgrammeItem to a Programmes-pages-service ProgrammeItem, why not just require the ProgrammeItem as part of the constructor for the ADA domain object.
* Most of our presenters present a single item. I think RelatedProgrammesPresenter should be RelatedProgramemPresenter and accept a single ADA domain object. Unless I'm missing something, which is very possible.
* There's a few todos and dump(); die(); kind of things that still need to be sorted.

**Pull request closed without merging**